server/src/test/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsResponseTests.java

```diff
@@ -49,4 +51,64 @@ public void testSerialization() throws IOException {
             }
         }
     }
+
+    @Override
+    protected GetFieldMappingsResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetFieldMappingsResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected GetFieldMappingsResponse createBlankInstance() {
+        return new GetFieldMappingsResponse();
+    }
+
+    @Override
+    protected GetFieldMappingsResponse createTestInstance() {
+        return new GetFieldMappingsResponse(randomMapping());
+    }
+
+    static int charCount(final String s, char ch) {
+        int count = 0;
+        int pos = 0;
+        while (pos >= 0) {
+            pos = s.indexOf(ch, pos);
+            if (pos >= 0) {
+                count++;
+                pos++;
+            }
+        }
+        return count;
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        // allow random fields at the level of index and index.mappings.doctype.field
+        // otherwise random field could be evaluated as index name or type name
+        return s -> {
+            final int c = s.length() > 0 ? charCount(s, '.') : -1;
+            return c != 0 && c != 3;};
```

Comment:
maybe a regex would be more readable here :)

Reply:
sure https://xkcd.com/208/